### PR TITLE
Fix: fallback to find when not a git repo

### PR DESCRIPTION
**Pull request by anryko**
__PROBLEM__
- `ansible-lint` is failing if run not in git repo.

__SOLUTION__
- Fallback to `find` in case if `git ls-files` fails.

### CODE REVIEW

File: lib/ansiblelint/utils.py [hunk 1]
```diff
@@ -763,15 +763,55 @@ def is_playbook(filename):
...
+    )
+
+    out, err = proc.communicate()
+
+    if err and options.verbosity:
+        print(
+            "Warning: Failed to discover yaml files to lint: %s" % (err)
+        )
+
+    if proc.returncode:
```
Comment (webknjaz) [inline lib/ansiblelint/utils.py]: Why don't you just use `subprocess.check_output()`? It seems cleaner.

File: lib/ansiblelint/utils.py [hunk 2]
```diff
@@ -763,15 +763,55 @@ def is_playbook(filename):
...
         options = {}
 
-    # git is preferred as it also considers .gitignore
-    files = OrderedDict.fromkeys(sorted(subprocess.check_output(
-        ["git", "ls-files", "*.yaml", "*.yml"],
-        universal_newlines=True).split()))
+    try:
+        # git is preferred as it also considers .gitignore
+        files = get_yaml_files(options=options)
+    except subprocess.CalledProcessError:
```
Comment (webknjaz) [inline lib/ansiblelint/utils.py]: Why don't you encapsulate the fallback logic in `get_yaml_files`?

File: lib/ansiblelint/utils.py [hunk 3]
```diff
@@ -763,15 +763,55 @@ def is_playbook(filename):
...
+
+    out, err = proc.communicate()
+
+    if err and options.verbosity:
+        print(
+            "Warning: Failed to discover yaml files to lint: %s" % (err)
+        )
+
+    if proc.returncode:
+        raise subprocess.CalledProcessError(
```
Comment (webknjaz) [inline lib/ansiblelint/utils.py]: LGTM linter is unhappy about this invocation.

**Comment by anryko** [comment]
@webknjaz thanks! I have Addressed your comments.

**Comment by ssbarnea** [comment]
If really needed we could improve the fallback mechanism to support non-git use-cases but we will need a detection mechanism and I am not sure who will have the time to write and maintain this code. Maybe it would make sense to propose a way to override the discovery command in the config file and we can avoid increasing the code size for uncommon use-cases.

**Comment by anryko** [comment]
@ssbarnea done

File: lib/ansiblelint/utils.py [hunk 4]
```diff
@@ -763,15 +763,48 @@ def is_playbook(filename):
...
+    try:
+        out = subprocess.check_output(
+            git_command,
+            stderr=subprocess.STDOUT,
+            universal_newlines=True
+        ).split()
+    except subprocess.CalledProcessError as exc:
+        if options.verbosity:
+            print(
+                "Warning: Failed to discover yaml files to lint using git: %s" % (
```
Comment (webknjaz) [inline lib/ansiblelint/utils.py]: Let's use `str.format`
```suggestion
                'Warning: Failed to discover yaml files to lint using git: {git_out!s}'.
                format(git_out=exc.output.rstrip('\n'))
```

File: lib/ansiblelint/utils.py [hunk 5]
```diff
@@ -763,15 +763,48 @@ def is_playbook(filename):
...
+    try:
+        out = subprocess.check_output(
+            git_command,
+            stderr=subprocess.STDOUT,
+            universal_newlines=True
+        ).split()
+    except subprocess.CalledProcessError as exc:
+        if options.verbosity:
+            print(
+                'Warning: Failed to discover yaml files to lint using git: {!s}'.format(
```
Comment (webknjaz) [inline lib/ansiblelint/utils.py]: Please use an indexed str placeholder.

File: lib/ansiblelint/utils.py [hunk 6]
```diff
@@ -763,15 +763,48 @@ def is_playbook(filename):
...
+def get_yaml_files(options=None):
+    """Find all yaml files."""
+    if options is None:
+        options = {}
+
+    # git is preferred as it also considers .gitignore
+    git_command = ['git', 'ls-files', '*.yaml', '*.yml']
+    if options.verbosity:
+        print(
+            "Discovering files to lint: %s" % (' '.join(git_command))
```
Comment (webknjaz) [inline lib/ansiblelint/utils.py]: plz use str format with an indexed placeholder here too.

File: lib/ansiblelint/utils.py [hunk 7]
```diff
@@ -763,15 +763,48 @@ def is_playbook(filename):
     return False
 
 
+def get_yaml_files(options=None):
```
Comment (webknjaz) [inline lib/ansiblelint/utils.py]: I think this shouldn't be optional as it's only used in one way in only one place making this assumption premature.

Comment (anryko) [reply lib/ansiblelint/utils.py]: Could you please elaborate?
Do you suggest to change this line to `def get_yaml_files(options):`?

Comment (webknjaz) [reply lib/ansiblelint/utils.py]: Exactly. Remove `=None` and the following `is None` if-clause block.

Comment (anryko) [reply lib/ansiblelint/utils.py]: done

**Comment by webknjaz** [comment]
Great, thanks!

**Pull request merged**